src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -164,17 +171,33 @@ public Buffer load(Key key)
         ByteBuffer buffer = bufferPool.get(key.file.chunkSize(), key.file.preferredBufferType());
         assert buffer != null;
         key.file.readChunk(key.position, buffer);
+        // Complete addition within compute remapping function to ensure there is no race condition with removal.
+        keysByFile.compute(key.internedPath, (k, v) -> {
```

Comment:
we need https://github.com/datastax/cassandra/commit/ef417d3356886fe9f6baf1f1fa697cf778f2e139 as well

Reply:
Sorry, I thought I had manually added those but I must have messed up the final merge. Force pushed.